### PR TITLE
Basic Docusarus Setup with Github Actions Deployments

### LINKED ISSUE #2: Set up Docusarus and Deploying to Github Pages
This should be a bare bones [Docusarus](https://docusaurus.io/) setup that deploys automatically on new commits to the `main` branch using Github Actions

Docusarus has a docs page about doing this which should be helpful: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

This task does NOT include creating a readme, there is another issue to track that work: https://github.com/BattlesnakeOfficial/docs-new/issues/1

### PR DESCRIPTION
**Pull request by coreyja**
This fixes #2

This was done with `create-docusarus` with the following command:
```bash
npx create-docusaurus@latest . classic
```

Also added the Github Actions for building on PRs and deploying on commits to main. They are basically copied from the docs here: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

## Decisions

I did have to make a few decisions here that I think could be worth discussing:

1. `npm` By default this is using `npm` while most of the Docusarus docs (and the readme before I edited it) mention `yarn`. I'm more used to yarn personally but don't have a strong preference. Do we want to switch and use `yarn` instead?
2. Typescript? Again by default this is not using Typescript, but the create-docusarus supports it. I thought it might make sense to use JS here to lower the burden for people contributing. But if there is want to do Typescript we can
3. Node Version. I picked `18.4.0` to use locally since that's the most recent `18.x` release, but can do 16 LTS or something if anyone has preferences

### CODE REVIEW

File: .github/workflows/test_build.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+      - name: Test build website
+        run: yarn run build
```
Comment (DaBultz) [inline .github/workflows/test_build.yml]: This is the only occurrence of `yarn`, shouldn't this have being `npm`?

**Comment by DaBultz** [comment]
> npm By default this is using npm while most of the Docusarus docs (and the readme before I edited it) mention yarn. I'm more used to yarn personally but don't have a strong preference. Do we want to switch and use yarn instead?

I would say, we should stick with `npm`. it comes pre-installed and setting up `yarn` with it's benefits are really a hit and miss, so I don't see any benefits `yarn` could give us. 

> Typescript? Again by default this is not using Typescript, but the create-docusarus supports it. I thought it might make sense to use JS here to lower the burden for people contributing. But if there is want to do Typescript we can

I love TypeScript, but it's more work. however, I don't see most people contributing with actual code, as the code is for custom components, functionality etc. not the actual content (markdown) which I see most people contribute to 9/10 of the time (if not more)

> Node Version. I picked 18.4.0 to use locally since that's the most recent 18.x release, but can do 16 LTS or something if anyone has preferences

Node 18 will be turned into LTS, 2022-10-25, so I would say, let's just stay on Node 18

File: .github/workflows/test_build.yml (hunk 1 above)
Comment (coreyja) [reply .github/workflows/test_build.yml]: Oh good catch! I even changed this line to add the `run` but missed the `yarn` 🤦

File: docusaurus.config.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  url: 'https://your-docusaurus-test-site.com',
+  baseUrl: '/',
+  onBrokenLinks: 'throw',
+  onBrokenMarkdownLinks: 'warn',
+  favicon: 'img/favicon.ico',
+
+  // GitHub pages deployment config.
+  // If you aren't using GitHub pages, you don't need these.
+  organizationName: 'facebook', // Usually your GitHub org/user name.
+  projectName: 'docusaurus', // Usually your repo name.
```
Comment (DaBultz) [inline docusaurus.config.js]: Looks like you skipped a step for the deployment to GH Pages, see link below
- https://docusaurus.io/docs/deployment#docusaurusconfigjs-settings

**Review by DaBultz — COMMENTED** [review]
Everything else besides the commented line looks okay to me, I assume that all branding will be done in another issue.

**Comment by 770grappenmaker** [comment]
> > npm By default this is using npm while most of the Docusarus docs (and the readme before I edited it) mention yarn. I'm more used to yarn personally but don't have a strong preference. Do we want to switch and use yarn instead?
> 
> I would say, we should stick with `npm`. it comes pre-installed and setting up `yarn` with it's benefits are really a hit and miss, so I don't see any benefits `yarn` could give us. 

Actually, I think since a certain nodejs version, yarn comes installed with nodejs, all you have to do is run some kind of corepack enable command:
https://yarnpkg.com/getting-started/install
So using yarn isn't a problem, I think

**Comment by DaBultz** [comment]
> > > npm By default this is using npm while most of the Docusarus docs (and the readme before I edited it) mention yarn. I'm more used to yarn personally but don't have a strong preference. Do we want to switch and use yarn instead?
> > 
> > I would say, we should stick with `npm`. it comes pre-installed and setting up `yarn` with it's benefits are really a hit and miss, so I don't see any benefits `yarn` could give us. 
> 
> Actually, I think since a certain nodejs version, yarn comes installed with nodejs, all you have to do is run some kind of corepack enable command:
> https://yarnpkg.com/getting-started/install
> So using yarn isn't a problem, I think

This is to install the new version (called berry), which broke a lot of things and packages break occasionally for yarn only, require special configuration and we have to disable features like "PnP" and we won't use many of the other features. 

That's being my experience with yarn and why i said hit and miss, as it's a mess.

I used to use yarn in the past (now called legacy), but now i end up only using it if there's a reason, otherwise i stick to npm.

There is pnpm, but i haven't used it yet, but i will still say that npm just sounds as the simplest and wouldn't cause friction.

**Review by jlafayette — APPROVED** [review]
Looks great!  Thanks for getting everything started

**Comment by jlafayette** [comment]
> npm By default this is using npm while most of the Docusarus docs (and the readme before I edited it) mention yarn. I'm more used to yarn personally but don't have a strong preference. Do we want to switch and use yarn instead?

I prefer npm just because I'm familiar with it, but I'm sure yarn would work fine too.

> Typescript? Again by default this is not using Typescript, but the create-docusarus supports it. I thought it might make sense to use JS here to lower the burden for people contributing. But if there is want to do Typescript we can

I'm ok with either one. I personally enjoy Typescript more but it doesn't really matter for smaller projects like this, so I'm fine sticking with JS for this.

> Node Version. I picked 18.4.0 to use locally since that's the most recent 18.x release, but can do 16 LTS or something if anyone has preferences

Using 18 sounds good to me!  We can always switch if a problem comes up.

File: docusaurus.config.js (hunk 2 above)
Comment (coreyja) [reply docusaurus.config.js]: This is what I get for not testing the code :facepalm: Thanks again!

**Pull request merged**